Sources/Defaults/SwiftUI.swift

```diff
@@ -6,25 +6,39 @@ extension Defaults {
 	final class Observable<Value: Serializable>: ObservableObject {
 		private var cancellable: AnyCancellable?
 		private var task: Task<Void, Never>?
-		private let key: Defaults.Key<Value>
-
-		let objectWillChange = ObservableObjectPublisher()
+		
+		var key: Defaults.Key<Value> {
+			didSet {
+				if oldValue != key {
+					observe()
+				}
+			}
+		}
 
 		var value: Value {
 			get { Defaults[key] }
 			set {
-				objectWillChange.send()
```

Comment:
I don't see any harm in keeping this? Unless I'm missing something. See: https://github.com/sindresorhus/Defaults/issues/144#issuecomment-1873194410

Reply:
If that is kept, the menubar will still freeze for some reason. Removing it fixes that, and doesn't seem to affect anything

Reply:
@Wouter01 Removing `objectWillChange.send()` broke animations when using `withAnimation`, so I will have to add it back unless you can think of a better solution.

Reproduction: [Fixture.zip](https://github.com/sindresorhus/Defaults/files/15469820/Fixture.zip)